kubespawner/spawner.py

```diff
@@ -3010,7 +3044,11 @@ async def load_user_options(self):
             )
 
     async def _ensure_namespace(self):
-        ns = make_namespace(self.namespace)
+        ns = make_namespace(
```

Comment:
Wait, shouldnt make namespace also be updated to support this?

Reply:
The fact that I discover this like this, even though tests pass, makes me think we should have a test for `make_namespace` like we do for many other functions, like `make_pod`.

Reply:
`make_namespace` already supports labels and annotations

https://github.com/jupyterhub/kubespawner/blob/616f72c4aee26c3d2127c6af6086ec50d6cda383/kubespawner/objects.py#L955

All that was needed was to add config options that would utilize this functionality